packages/analytics/dashboard-renderer/src/components/layout/GridLayout.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <div class="kong-ui-public-grid-layout">
+    <div
+      v-for="cell in totalCells"
+      :key="cell.key"
+      class="grid-cell"
+      :style="{
+        gridRow: `span ${cell.tile?.size.rows}`,
+        gridColumn: `span ${cell.tile?.size.cols}`,
+      }"
+    >
+      <slot
+        v-if="cell.tile"
+        class="tile-content"
+        name="tile"
+        :tile="cell.tile"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { defineProps, computed, type PropType } from 'vue'
+import type { GridSize, Tile } from 'src/types'
+
+interface Cell {
+  key: string
+  tile?: Tile
+}
+
+const props = defineProps({
+  gridSize: {
+    type: Object as PropType<GridSize>,
+    required: true,
+  },
+  tileHeight: {
+    type: Number,
+    required: false,
+    default: 150,
+  },
+  tileWidth: {
+    type: Number,
+    required: false,
+    default: 150,
+  },
+  tiles: {
+    type: Array as PropType<Tile[]>,
+    required: true,
+  },
+})
+
+const totalCells = computed<Cell[]>(() => {
+  const cells: Cell[] = []
+  const occupied = new Set()
+
+  for (let row = 1; row <= props.gridSize.rows; row++) {
+    for (let col = 1; col <= props.gridSize.cols; col++) {
+      const key = `row-${row}-col-${col}`
+
+      // Skip cell creation if the position is already occupied by a spanning tile
+      if (occupied.has(key)) continue
+
+      const tile = props.tiles.find(t => t.position.row === row && t.position.col === col)
+      cells.push({
+        key,
+        tile,
+      })
+
+      // If the cell has a tile that spans multiple rows/cols,
+      // mark all spanned positions as occupied
+      if (tile) {
+        for (let i = 0; i < tile.size.rows; i++) {
+          for (let j = 0; j < tile.size.cols; j++) {
+            occupied.add(`row-${row + i}-col-${col + j}`)
+          }
+        }
+      }
+    }
+  }
```

Comment:
What's the performance here look like? You have two sets of nested loops within the a parent loop

Reply:
I'm assuming we're not going to have massive grids with thousands of cells, with tiles spanning 100s of cells. 

We can put a hard limit on the size of grid we will have to prevent performance issues.

The tile rows/cols nested loop will only loop for how many cells that tile spans, which should never be more than a few col,row spans

Reply:
I still have some concerns with the loops 😬

Reply:
Ok, reduced the number of loops, now it's just one nested loop.